app/Services/Campaign/MemberService.php

```diff
@@ -42,20 +42,23 @@ public function fromRequest(UpdateUserRole $request): self
 
     /**
      */
-    public function update(CampaignUser $user, CampaignRole $campaignRole): bool
+    public function update(CampaignUser $user, array $roles): bool
     {
+        $currentRoles = $user->roles->pluck('id')->toArray();
+        $roleUsers = CampaignRoleUser::where('user_id', $user->user_id)
+            ->whereIn('campaign_role_id', $currentRoles)
+            ->whereNotIn('campaign_role_id', $roles)
+            ->with('campaignRole')
+            ->get();
+
+        $deletedRoles = [];
         /** @var ?CampaignRoleUser $role */
-        $role = CampaignRoleUser::where('user_id', $user->user_id)
-            ->where('campaign_role_id', $campaignRole->id)
-            ->first();
-
-        // Admin role being switched? Forget the cache
-        if ($campaignRole->isAdmin()) {
-            CampaignCache::campaign($campaignRole->campaign)->clear();
-        }
-
-        // Deleting an existing role
-        if ($role) {
+        foreach ($roleUsers as $role) {
```

Comment:
I feel like the part where an admin can't remove another admin after the first 15 minutes that they've been added to the admin role (or something like that) is missing here :P